test/units/testsuite-21.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+set -eux
+set -o pipefail
+
+# Save the end.service state before we start fuzzing, as it might get changed
+# on the fly by one of the fuzzers
+systemctl is-enabled end.service && SHUTDOWN_AT_EXIT=1 || SHUTDOWN_AT_EXIT=0
+
+at_exit() {
+    # "Safety net" - check for any coredumps which might have not caused dfuzzer
+    # to stop & return an error (we need to do this now before truncating the
+    # journal)
+    # TODO: check fo ASan/UBSan errors
+    local found_cd=0
+    while read -r exe; do
+        coredumctl info "$exe"
+        found_cd=1
+    done < <(coredumpctl -F COREDUMP_EXE | sort -u)
+    [[ $found_cd -eq 0 ]] || exit 1
+
+    # Limit the maximum journal size (because fuzzing)
+    journalctl --rotate --vacuum-size=16M
```

Comment:
I think if the idea is to avoid running out of space the journal should probably be trimmed on every dfuzzer run. I mean, if the test gets to this point it can safely keep the journal to make it possible to extract coredumps and ASan backtraces as usual and delete it after that.